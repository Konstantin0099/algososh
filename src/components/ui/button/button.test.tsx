
import React from "react";
import { Button } from "./button";
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from "react-test-renderer";
import styles from "./main-page.module.css";

const onClick = () => {
  alert('buttonClick')
}

describe('+++Button+++', () => {
  
  it("!!!!!with text Button", () => {
  const buttonSnapshot = renderer
  .create(<Button text="textInButton"/>)
  .toJSON();
  expect(buttonSnapshot).toMatchSnapshot();
});
it("!!!!!without text Button", () => {
  const buttonSnapshot = renderer
  .create(<Button />)
  .toJSON();
  expect(buttonSnapshot).toMatchSnapshot();
});
it("!!!!!!disabled Button", () => {
  const buttonSnapshot = renderer
  .create(<Button disabled={true} />)
  .toJSON();
  expect(buttonSnapshot).toMatchSnapshot();
});
it("!!!!!!disabled Button", () => {
  // render(<Button text="textInButton"/>);
  // const buttonWithText = screen.getByText("textInButton");
  const buttonSnapshot = renderer
  .create(<Button isLoader={true} />)
  .toJSON();
  expect(buttonSnapshot).toMatchSnapshot();
  // expect(buttonWithText).toBeInTheDocument();
});
it('Нажатие на кнопку вызывает корректный alert', () => {
  window.alert = jest.fn();
  render(<Button text="textInButton" onClick={onClick}/>);
  const buttonWithText = screen.getByText("textInButton");
  fireEvent.click(buttonWithText);
  expect(window.alert).toHaveBeenCalledWith('buttonClick');
}); 

})