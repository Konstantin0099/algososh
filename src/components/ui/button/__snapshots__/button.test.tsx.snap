// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`+++Button+++ !!!!!!disabled Button 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`+++Button+++ !!!!!!disabled Button 2`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`+++Button+++ !!!!!with text Button 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    textInButton
  </p>
</button>
`;

exports[`+++Button+++ !!!!!without text Button 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
